std/js/html/DOMElement.hx

```diff
@@ -162,95 +284,264 @@ extern class DOMElement extends Node
-	/** @throws DOMError */
 	
 	/**
-		Designates a specific element as the capture target of future `PointerEvent`.
+		Inserts a given element node at a given position relative to the element it is invoked upon.
+		@throws DOMError
+	**/
+	@:pure
+	function insertAdjacentElement( where : String, element : Element ) : Element;
+	
+	/**
+		Inserts a given text node at a given position relative to the element it is invoked upon.
+		@throws DOMError
+	**/
+	function insertAdjacentText( where : String, data : String ) : Void;
+	
+	/**
+		Designates a specific element as the capture target of future pointer events.
+		@throws DOMError
 	**/
 	function setPointerCapture( pointerId : Int ) : Void;
-	/** @throws DOMError */
+	
+	/**
+		Releases (stops) pointer capture that was previously set for a specific `PointerEvent`.
+		@throws DOMError
+	**/
 	function releasePointerCapture( pointerId : Int ) : Void;
+	function hasPointerCapture( pointerId : Int ) : Bool;
+	
+	/**
+		Sets up mouse event capture, redirecting all mouse events to this element.
+	**/
 	function setCapture( ?retargetToElement : Bool = false ) : Void;
 	function releaseCapture() : Void;
-	function requestPointerLock() : Void;
+	
+	/**
+		Retrieves the node representation of the named attribute from the current node and returns it as an `Attr`.
+	**/
 	function getAttributeNode( name : String ) : Attr;
-	/** @throws DOMError */
+	
+	/**
+		Sets the node representation of the named attribute from the current node.
+		@throws DOMError
+	**/
 	function setAttributeNode( newAttr : Attr ) : Attr;
-	/** @throws DOMError */
+	
+	/**
+		Removes the node representation of the named attribute from the current node.
+		@throws DOMError
+	**/
 	function removeAttributeNode( oldAttr : Attr ) : Attr;
+	
+	/**
+		Retrieves the node representation of the attribute with the specified name and namespace, from the current node and returns it as an `Attr`.
+	**/
 	function getAttributeNodeNS( namespaceURI : String, localName : String ) : Attr;
-	/** @throws DOMError */
+	
+	/**
+		Sets the node representation of the attribute with the specified name and namespace, from the current node.
+		@throws DOMError
+	**/
 	function setAttributeNodeNS( newAttr : Attr ) : Attr;
 	function click() : Void;
 	/** @throws DOMError */
 	function focus() : Void;
 	/** @throws DOMError */
 	function blur() : Void;
+	
+	/**
+		Returns a collection of rectangles that indicate the bounding rectangles for each line of text in a client.
+	**/
 	function getClientRects() : DOMRectList;
+	
+	/**
+		Returns the size of an element and its position relative to the viewport.
+	**/
 	function getBoundingClientRect() : DOMRect;
-	@:overload( function( top : Bool ) : Void {} )
-	function scrollIntoView( ?options : ScrollIntoViewOptions ) : Void;
+	
+	/**
+		Scrolls the page until the element gets into the view.
+	**/
+	@:overload( function( ?arg : ScrollIntoViewOptions) : Void {} )
+	function scrollIntoView( ?arg : Bool ) : Void;
```

Comment:
I'm late to the party, but this change is problematic.

Having `?arg : Bool` instead of `?arg : ScrollIntoViewOptions` as the primary signature for `scrollIntoView` breaks type inference and makes the use of `ScrollIntoViewOptions` really annoying.

Is there a way to prioritize in the generator?

Reply:
Relying on type-inference on overloaded functions is asking for trouble. But if this is a case where we flipped the default then it might make sense to stick with the previous order.

Reply:
Are we supposed to `ECheckType` every "object" argument in overloaded functions? 

This seems really annoying to use =/
```haxe
el.scrollIntoView(({behavior: SMOOTH, block: NEAREST} :js.html.ScrollIntoViewOptions));
```

Reply:
The order comes from the WebIDLs but I could try to come up with a heuristic to pick the 'best' primary signature

Reply:
But that is already typed. How is type-inference even involved here? A small example would probably help.

Reply:
Sorry, it's probably a vocabulary issue from me. And also some wrong guesses (animate seems fine for example).

I cannot do this:
`el.scrollIntoView({behavior: SMOOTH, block: NEAREST});` without adding the `ECheckType`, I get an `Unknown identifier : SMOOTH` error.

Reply:
Ah I see. That's probably something we should try to fix in the compiler, but I'm not sure if that's really possible with our overload architecture. Please open an issue about this, preferable with an example that doesn't require any JS externs.

Reply:
Ok, will do!

Reply:
In the most recent versions of the compiler, `EitherType` seems to be better at inference (resolves this specific case) so I could switch to that in the externs

However I guess the best solution would be to somehow have the same level of inference for overloads

(I actually initially switched to overloads for these cases because it had better type inference than `EitherType`)